src/ncp-spinel/SpinelNCPControlInterface.cpp

```diff
@@ -309,6 +310,62 @@ SpinelNCPControlInterface::add_external_route(
 }
 
 void
+SpinelNCPControlInterface::joiner_add(
+		const uint8_t *addr,
+		int addr_len,
+		const char *psk,
+		int psk_len,
+		CallbackWithStatus cb
+) {
+
+	require_action(addr != NULL, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk != NULL, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk_len >= PSK_MIN_LENGTH, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(psk_len <= PSK_MAX_LENGTH, bail, cb(kWPANTUNDStatus_InvalidArgument));
+	require_action(mNCPInstance->mEnabled, bail, cb(kWPANTUNDStatus_InvalidWhenDisabled));
+
+	mNCPInstance->start_new_task(SpinelNCPTaskSendCommand::Factory(mNCPInstance)
+		.set_callback(cb)
+		.add_command(SpinelPackData(
+			SPINEL_FRAME_PACK_CMD_PROP_VALUE_INSERT(
+			SPINEL_DATATYPE_EUI64_S
+			SPINEL_DATATYPE_DATA_S
+			SPINEL_DATATYPE_UINT8_S
+		),
+		SPINEL_PROP_THREAD_JOINERS,
+		addr,
+		psk,
+		psk_len
+	))
+	.set_lock_property(SPINEL_PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE)
+	.finish()
+	);
+
+bail:
+	return;
+}
+
+void
+SpinelNCPControlInterface::commissioner(
```

Comment:
Suggestion: Since this is just to enable/disable commissioner operation, I think it may be easier if instead of adding it as a `NCPControlInterface`  new method, we define it as a wpan property (which is then tied to the spinel property... `SPINEL_PROP_THREAD_COMMISSIONER_ENABLED`)...Thoughts?